kit/torch/data.py

```diff
@@ -26,3 +29,53 @@ def prop_random_split(
         section_sizes.append(len_ - sum(section_sizes))
     generator = torch.default_generator if seed is None else torch.Generator().manual_seed(seed)
     return random_split(dataset, section_sizes, generator=generator)
+
+
+class InfSequentialSampler(Sampler[List[int]]):
+    r"""Infinitely samples elements sequentially, always in the same order.
+    This is useful for enabling iteration-based training.
+
+    Args:
+        data_source (Sized): dataset to sample from
+    """
+
+    def __init__(self, data_source: Sized, batch_size: int, shuffle: bool = True) -> None:
+        self.data_source = data_source
+        self.shuffle = shuffle
+        self._dataset_size = len(data_source)
+        self.batch_size = min(batch_size, self._dataset_size)
+
+    def _generate_idx_seq(self) -> Tensor:
+        """Generate a random sequence of unique indexes."""
+        if self.shuffle:
+            return torch.arange(self._dataset_size)
+        return torch.randperm(self._dataset_size)
+
+    def batch_indexes(self, indexes: Tensor) -> Sequence[Tensor]:
+        """Split the indexes into batches."""
+        return indexes.split(self.batch_size)
+
+    @implements(Sampler)
+    def __iter__(self) -> Iterator[List[int]]:
+        batched_idxs_iter = iter(self.batch_indexes(self._generate_idx_seq()))
+        # Iterate until some externally-defined stopping criterion is reached
+        while True:
+            batch_idxs = next(batched_idxs_iter, None)  # type: ignore
+            if batch_idxs is None or (len(batch_idxs) < self.batch_size):
+                new_idx_seq = self._generate_idx_seq()
+                if batch_idxs is not None:
+                    # Rather than dropping the last batch if it is incomplete or simply using it,
+                    # incomlpete as it may be, we take the alternative approach of concatenating the surplus
```

Comment:
No big deal, but might as well fix it

```suggestion
                    # incomplete as it may be, we take the alternative approach of concatenating the surplus
```